pydantic/types.py

```diff
@@ -479,15 +479,17 @@ class ConstrainedList(list):  # type: ignore
 
     min_items: Optional[int] = None
     max_items: Optional[int] = None
+    unique_items: Optional[bool] = None
     item_type: Type[T]  # type: ignore
 
     @classmethod
     def __get_validators__(cls) -> 'CallableGenerator':
         yield cls.list_length_validator
+        yield cls.unique_items_validator
```

Comment:
```suggestion
        if cls.unique_items:
            yield cls.unique_items_validator
```
This should mean the check can occur outside performance sensitive validation stage.

Reply:
It's the first checking in the validator but this a better place, indeed.